src/main/java/tutoraid/model/student/InitialStudent.java

```diff
@@ -67,7 +67,7 @@ public void addStudentLessonLink(String lessonName, List<Lesson> fullLessonList,
             throws IOException {
         for (Lesson lesson : fullLessonList) {
             if (lesson.toNameString().equals(lessonName) && lesson.isFull()) {
-                lesson.removeAllStudents();
+                fullLessonList.forEach(Lesson::removeAllStudents);
```

Comment:
This change results in `Alex Yeoh` not appearing in the `Students` field under the lesson `Maths 1` when the app is first launched. The name appears on subsequent launches though. Trying to find out why and also any possible fixes to this, since this change helps to solve the JSON over-capacity issue, so we cannot fix it by reverting this change

Reply:
Oh wait I found out this bug still occurs when this change is reverted. So, this bug does not stem from this change. Trying to figure out the real reason why this bug occurs